src/sdam/server_description.ts

```diff
@@ -111,6 +117,9 @@ export class ServerDescription {
     this.setVersion = hello?.setVersion ?? null;
     this.electionId = hello?.electionId ?? null;
     this.logicalSessionTimeoutMinutes = hello?.logicalSessionTimeoutMinutes ?? null;
+    this.maxMessageSizeBytes = hello?.maxMessageSizeBytes ?? 0;
```

Comment:
I don't know if setting it to zero is the correct default - I think `null` to indicate "not provided" is preferable.  `0` could cause other parts of the driver to behave strangely whereas null will force users of this data to check is its defined and handle the case where its not explicitly.

Although, I wish there were a way we could make these fields required on the `hello` somehow.  I'd be surprised if they're ever undefined on our supported server versions.

Reply:
The reason I defaulted to 0 here instead of null is that the end result is going to be the same but without having the extra checks for existence. If any of these values is either 0 or null no messages will be able to be added - and since these are new properties on the server description they haven't been used anywhere else as of yet. Let me know if you feel strongly or not about it.

Reply:
Per conversation we've defaulted to null and will throw later if they are actually null when building the command batches.